src/include/btree_inline.h

```diff
@@ -2352,6 +2344,8 @@ __wt_btcur_skip_page(
      * if the page has been modified since it was reconciled, since neither the delete information
      * nor the timestamp information is necessarily up to date.
      */
+    WT_ASSERT_ALWAYS(session, __wt_session_gen(session, WT_GEN_SPLIT) != 0,
```

Comment:
Shouldn't we build this assertion into `__wt_ref_addr_copy`? Also the comment on this assertion confuses me as it is copying an addr not freeing it.

Reply:
No. The `__wt_ref_addr_copy` is called from locations where the split generation is not acquired. Due to this reason, we cannot move this assert into that function. Otherwise, all the callers have to take the split generation when calling this function.

Reply:
I think we should require all callers to take the relevant generation, if that requires creating a WT_REF_ADDR generation then I'd be okay with that. Otherwise this is a half-baked usage of generations where some functions require that semantic and others don't but there's no obvious distinction between the tow.

Reply:
Thanks. I find there are not many places where we need to add the split generation. So I went ahead with WT_GEN_SPLIT in the callers.

Reply:
Thanks Hari I think that makes this a lot easier to reason about!